### PR TITLE
Don't fail the build on invalid Problems API invocation

### LINKED ISSUE #27353: Make Problems API more permissive
The current state only changes the tooling API behavior. Consequently it does not make sense to fail the build if something is misconfigured (i.e. label or category is missing from the problem configuration). Instead, we should report a separate problem about missing configuration with `stacklocation`.

### PR DESCRIPTION
**Pull request by donat**
Fixes #27353

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test QFL

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test RFM

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -16,501 +16,324 @@
...
+    }
+
+    Map<String, Object> getCollectedProblem() {
+        assert this.collectedProblems.size() == 1
+        this.collectedProblems[0]
+    }
+
+    def "problem replaced with a validation warning if mandatory label definition is missing"() {
+        given:
+        withReportProblemTask"""
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:53]: 💅 
```suggestion
        withReportProblemTask """
```

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 2]
```diff
@@ -16,501 +16,324 @@
...
-
-                @TaskAction
-                void run() {
-                    problems.forNamespace("org.example.plugin").reporting {
-                        it.label("label")
-                        .documentedAt("https://example.org/doc")
-                        .category("type")
-                    }
-                }
+        withReportProblemTask"""
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:70]: 💅 
```suggestion
        withReportProblemTask """
```

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 3]
```diff
@@ -16,501 +16,324 @@
...
 
-    def "can emit a problem with upgrade-guide documentation"() {
+
+    def "can emit a problem with minimal configuration"() {
         given:
-        buildFile """
-            import org.gradle.api.problems.internal.Problem
-            import org.gradle.api.problems.Severity
-            import org.gradle.internal.deprecation.Documentation
+        withReportProblemTask"""
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:88]: 💅 
```suggestion
        withReportProblemTask """
```

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 4]
```diff
@@ -16,501 +16,324 @@
...
-                        }
-                    } catch (RuntimeException ex) {
-                        problems.forNamespace("org.example.plugin").rethrowing(ex) { spec -> spec
-                            .label("outer")
-                            .category("type")
-                        }
-                    }
+        withReportProblemTask"""
+            try {
+                def exception = new RuntimeException("test")
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:321]: 💅 I think we should inline `exception`

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 5]
```diff
@@ -16,501 +16,324 @@
...
-                @TaskAction
-                void run() {
-                    def exception = new RuntimeException("test")
-                    problems.forNamespace("org.example.plugin").rethrowing(exception) { it
-                        .label("label")
-                        .category("type")
-                    }
-                }
+        withReportProblemTask"""
+            def exception = new RuntimeException("test")
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:303]: 💅 I think we should inline `exception`

**Review by reinsch82 — CHANGES_REQUESTED** [review]
A few small things.
note: I didn't mark all the missing whitespaces for `withReportProblemTask`

**Comment by donat** [comment]
@bot-gradle test RFM

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy (hunk 1 above)
Comment (donat) [reply platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy]: Fixed by reformatting the code.

**Review by reinsch82 — APPROVED** [review]
LGTM

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**